internal/pkg/agent/application/managed_mode.go

```diff
@@ -194,12 +197,35 @@ func (m *managedConfigManager) Run(ctx context.Context) error {
 		}
 	})
 
+	go func() {
+		for {
+			select {
+			case <-ctx.Done():
+				return
+			case err := <-actionDispatcher.Errors():
+				m.errCh <- err // err is one or more failures from dispatching an action
+			}
+		}
+	}()
+
 	// Run the gateway.
 	gatewayRunner := runner.Start(gatewayCtx, func(ctx context.Context) error {
 		defer gatewayErrorsRunner.Stop()
 		return gateway.Run(ctx)
 	})
 
+	// pass actions collected from gateway to dispatcher
+	go func() {
+		for {
+			select {
+			case <-ctx.Done():
+				return
+			case actions := <-gateway.Actions():
+				actionDispatcher.Dispatch(ctx, actionAcker, actions...)
```

Comment:
I like this design much better. Separating the 2 and then connect them together in the manager.